cmd/libsnap-confine-private/cgroup-support.c

```diff
@@ -133,11 +140,17 @@ static bool traverse_looking_for_prefix_in_dir(DIR *root, const char *prefix, co
         // entfd is consumed by fdopendir() and freed with closedir()
         int entfd = openat(dirfd(root), ent->d_name, O_DIRECTORY | O_NOFOLLOW | O_CLOEXEC);
         if (entfd == -1) {
+            if (errno == ENOENT) {
+                // the processes may exit and the group entries may go away at
+                // any time
+                return false;
+            }
             die("cannot open directory entry \"%s\"", ent->d_name);
         }
         // takes ownership of the file descriptor
         DIR *entdir SC_CLEANUP(sc_cleanup_closedir) = fdopendir(entfd);
         if (entdir == NULL) {
+            // we have the fd, so ENOENT isn't possible here
```

Comment:
```
g_assert(errno != ENOENT);
```
would reinforce this statement :-)

Reply:
mmm I don't think we use g_assert at all in runtime code, only for test code

Reply:
Yeah, we don't link to glib in the production code to minimize the possible attack surface of this suid root binary.

Reply:
(having glib would be soooooooooooo nice compared to plain old C fwiw)

Reply:
we also need to link statically, AFAIU glib does not encourage using static builds of the library (if building it this way is even still possible)

Reply:
What about plain `assert()`? Could this be used instead?

Reply:
AFAIK g_assert() expands to nothing in production code, so we wouldn't actually be linking to it. But I agree that it could be confusing have `g_*` things in here, because it would give the reader the impression that using glib functions is accepted.
As Alex suggests, using `assert()` should solve this too.